pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -161,40 +163,39 @@ func (c *Controller) updateOperatorStatus(ctx context.Context, previousStatus *o
 		errs = []error{}
 	}
 
-	deploymentAvailableCondition := operatorv1.OperatorCondition{
-		Type:   fmt.Sprintf("%sDeployment%s", c.conditionsPrefix, operatorv1.OperatorStatusTypeAvailable),
-		Status: operatorv1.ConditionTrue,
-	}
+	deploymentAvailableCondition := applyoperatorv1.OperatorCondition().
+		WithType(fmt.Sprintf("%sDeployment%s", c.conditionsPrefix, operatorv1.OperatorStatusTypeAvailable))
 
-	workloadDegradedCondition := operatorv1.OperatorCondition{
-		Type:   fmt.Sprintf("%sWorkloadDegraded", c.conditionsPrefix),
-		Status: operatorv1.ConditionFalse,
-	}
+	workloadDegradedCondition := applyoperatorv1.OperatorCondition().
+		WithType(fmt.Sprintf("%sWorkloadDegraded", c.conditionsPrefix))
 
-	deploymentDegradedCondition := operatorv1.OperatorCondition{
-		Type:   fmt.Sprintf("%sDeploymentDegraded", c.conditionsPrefix),
-		Status: operatorv1.ConditionFalse,
-	}
+	deploymentDegradedCondition := applyoperatorv1.OperatorCondition().
+		WithType(fmt.Sprintf("%sDeploymentDegraded", c.conditionsPrefix))
+
+	deploymentProgressingCondition := applyoperatorv1.OperatorCondition().
+		WithType(fmt.Sprintf("%sDeployment%s", c.conditionsPrefix, operatorv1.OperatorStatusTypeProgressing))
 
-	deploymentProgressingCondition := operatorv1.OperatorCondition{
-		Type:   fmt.Sprintf("%sDeployment%s", c.conditionsPrefix, operatorv1.OperatorStatusTypeProgressing),
-		Status: operatorv1.ConditionFalse,
+	status := applyoperatorv1.OperatorStatus()
+	if workload != nil {
+		status = status.WithGenerations(applyoperatorv1.GenerationStatus().
```

Comment:
The previous [function](https://github.com/openshift/library-go/blob/816295c705d36a0115f4dc4340c94ebbc227158d/pkg/operator/resource/resourcemerge/apps.go#L37) was also setting `Hash` field. Not needed anymore ?

Reply:
> he previous [function](https://github.com/openshift/library-go/blob/816295c705d36a0115f4dc4340c94ebbc227158d/pkg/operator/resource/resourcemerge/apps.go#L37) was also setting `Hash` field. Not needed anymore ?

Never set, rigth? https://github.com/openshift/library-go/blob/816295c705d36a0115f4dc4340c94ebbc227158d/pkg/operator/resource/resourcemerge/apps.go#L48-L59

Reply:
set in the SetGeneration function (https://github.com/openshift/library-go/blob/816295c705d36a0115f4dc4340c94ebbc227158d/pkg/operator/resource/resourcemerge/apps.go#L37)

Reply:
So always empty since the newGeneration never had it set, rigth?

Reply:
Previously, this controller set the generation using the `SetGenerationDeployment` helper, which doesn't set the hash field:

https://github.com/openshift/library-go/blob/master/pkg/operator/apiserver/controller/workload/workload.go#L316-L320

This helpers calls `SetGeneration`, which does copy the field over if it exists, but in practice the field is always empty because `SetGeneration` is called through `SetDeploymentGeneration`.

Example:

```
omc get authentications.operator.openshift.io/cluster -o json | jq .status.generations
[
  {
    "group": "apps",
    "hash": "",
    "lastGeneration": 3,
    "name": "apiserver",
    "namespace": "openshift-oauth-apiserver",
    "resource": "deployments"
  },
  {
    "group": "apps",
    "hash": "",
    "lastGeneration": 2,
    "name": "oauth-openshift",
    "namespace": "openshift-authentication",
    "resource": "deployments"
  }
]
```

Reply:
Added a comment to explain why the field is not required for a deployment